src/learning-catalogue/model/factory/moduleFactory.ts

```diff
@@ -12,17 +12,19 @@ export class ModuleFactory {
 	constructor(eventFactory: EventFactory = new EventFactory()) {
 		this.eventFactory = eventFactory
 
-		this.defaultCreate = this.defaultCreate.bind(this)
 		this.create = this.create.bind(this)
 	}
 
-	public defaultCreate(module: any, data: any) {
+	private static defaultCreate(
+		module: VideoModule | LinkModule | FileModule | FaceToFaceModule | ELearningModule,
+		data: any
+	) {
 		module.id = data.id
 		module.type = data.type
 		module.title = data.title
 		module.description = data.description
 		module.duration = data.duration
-		module.price = data.price
+		module.cost = !isNaN(Number(data.cost)) ? Number(data.cost) : data.cost
```

Comment:
Isn't cost only associated with Face-to-Face modules?

Reply:
If it is in business domain, it's not in technical terms - `learning-catalogue` has got `cost` on `Module` class.